packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -216,17 +216,24 @@ class DateTimePicker extends React.Component {
 
     let updatedDateTime;
     const formattedDate = DateTimeUtils.formatISODateTime(date, 'YYYY-MM-DD');
+    const isDateValid = DateTimeUtils.isValidDate(formattedDate, 'YYYY-MM-DD');
+    const isTimeValid = DateTimeUtils.isValidTime(this.timeValue);
 
-    if (DateTimeUtils.isValidDate(formattedDate, 'YYYY-MM-DD')) {
-      const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    if (isDateValid) {
+      const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : DateUtil.createSafeDate(formattedDate);
       updatedDateTime = DateTimeUtils.syncDateTime(previousDateTime, date, this.timeValue);
 
-      if (DateTimeUtils.isValidTime(this.timeValue)) {
+      if (isTimeValid) {
+        // Update the timeValue in case the updatedDateTime falls in the missing hour and needs to bump the hour up.
         this.timeValue = DateTimeUtils.formatISODateTime(updatedDateTime.format(), 'HH:mm');
       }
     }
 
-    this.handleChange(event, updatedDateTime);
+    if ((isDateValid && isTimeValid) || (this.dateValue === '' && this.timeValue === '')) {
```

Comment:
May be good to add a comment here clarifying why were adding this logic.

Reply:
Added comment: https://github.com/cerner/terra-framework/pull/619/commits/0fdc394ea8acf93005f45918763597ecb9d98c90